src/Microsoft.DotNet.Cli.Utils/CommandResolution/PackagedCommandSpecFactory.cs

```diff
@@ -23,25 +19,25 @@ public class PackagedCommandSpecFactory : IPackagedCommandSpecFactory
             CommandResolutionStrategy commandResolutionStrategy,
             string depsFilePath)
         {
-            var packageDirectory = GetPackageDirectoryFullPath(library, nugetPackagesRoot);
 
-            if (!Directory.Exists(packageDirectory))
+            var toolAssembly = toolLibrary?.RuntimeAssemblies
+                    .FirstOrDefault(r => Path.GetFileNameWithoutExtension(r.Path) == commandName);
```

Comment:
Talked offline, not checking the extension doesn't break any of our current scenarios.